### ISSUE TITLE
Sets are constructed with only one member

### ISSUE THREAD

**Issue by akutschera** [Nylle/JavaFixture#102]
Any non-primitive set will only contain one member.

The following test fails
```
    @TestWithFixture(minCollectionSize = 2, maxCollectionSize = 2)
    void setTest( Set<Object> sut ) {
        assertThat( sut ).hasSize( 2 );
    }
```
because the set size is only one.
I know this is a problem because internally JavaFixture is caching previously created objects for re-use.
If we found a way to solve this, this could also mean that Lists could (will?) contain different members, which is a more realistic scenario, I guess.

**Comment by akutschera** [comment]
With the above commits, we should only keep specimens in the cache for circular dependencies. After a specimen is created, it is removed from the cache.

**Comment by Nylle** [comment]
Great solution!

**Comment by Nylle** [comment]
(Will be resolved as fixed after release.)

**Comment by jk-idealo** [comment]
Fixed since version 2.11.0.
